Add post about GOV.UK Eleventy Plugin (#14)
x-govuk/x-govuk.github.io: app/posts/govuk-eleventy-plugin.md

Markdown written using [an extended syntax](https://x-govuk.github.io/govuk-eleventy-plugin/markdown-advanced/) is also supported, meaning authors can easily include tables, abbreviations, footnotes and code examples in their posts, too.

Knowing that teams often need to write documentation for internal users, or build websites that don't meet [the criteria for using the GOV.UK logo and GDS Transport typeface](https://www.gov.uk/service-manual/design/making-your-service-look-like-govuk), the plugin enables the following can be customised:

* Organisation name and logo
* Font family
* Brand colour
* Licence and copyright information
* Site icons and Open Graph share images

![Examples of customised websites using the plugin.](/assets/posts/govuk-eleventy-plugin/customisation-examples.png)

## Get started

The plugin is already being used by different teams across government who have built marketing pages, blogs, glossaries – and design histories, of course.

> One of the benefits of static websites is they don't need a database, which means they can be hosted pretty much anywhere. Popular platforms for hosting static websites include [GitHub Pages](https://pages.github.com) and [Netlify](https://www.netlify.com).

Author: @frankieroberto Possibly a bit too late in the post, but reworded this section to add a bit of colour about what a static website is.

---
title: Write documentation using Markdown and publish it using GOV.UK styles
description: The GOV.UK Eleventy Plugin makes it easy to start writing documentation rather than spend time building a website for it.
date: 2022-04-04
tags: post
layout: post
author:
  name: Paul Lloyd
  url: https://github.com/paulrobertlloyd
image:
  src: /assets/posts/govuk-eleventy-plugin/govuk-eleventy-plugin.png
  alt: The homepage illustration for the GOV.UK Eleventy Plugin.
---

> Back in 2018, the Becoming a teacher team at the Department for Education wanted to [document the design evolution of the services they were building](https://design-history.herokuapp.com/keeping-a-design-history/). A small site was built using the [GOV.UK Prototype Kit](https://govuk-prototype-kit.herokuapp.com/docs), but this soon became difficult to manage as more people joined the team and wanted to write posts.

Reviewer: ```suggestion
Back in 2018, the Becoming a teacher team at the Department for Education wanted to [document the design evolution of the services they were building](https://design-history.herokuapp.com/keeping-a-design-history/). A small website was built using the [GOV.UK Prototype Kit](https://govuk-prototype-kit.herokuapp.com/docs), but this soon became difficult to manage as more people joined the team and wanted to write posts.
```

> Looking for alternatives, the team choose [Eleventy](https://11ty.dev), a simple static site generator developed by Zach Leatherman[^1]. Like the Prototype Kit, it uses Node.js, yet is designed around organising and publishing content-focused websites.

Reviewer: Minor typo:

```suggestion
Looking for alternatives, the team chose [Eleventy](https://11ty.dev), a simple static site generator developed by Zach Leatherman[^1]. Like the Prototype Kit, it uses Node.js, yet is designed around organising and publishing content-focused websites.
```

I also wonder whether you could briefly explain what a “static site generator” is here, for people not familiar with the term?

> This includes support for [Markdown](https://www.markdownguide.org), a lightweight markup language for formatting plain text documents. Posts can incorporate additional data like publication dates and authors, and be organised into different collections.

Reviewer: ```suggestion
This includes support for [Markdown](https://www.markdownguide.org), a lightweight markup language for formatting plain text documents. Posts can incorporate metadata like publication dates and authors, and be organised into different collections.
```

> With this foundation in place, [the design history](https://bat-design-history.netlify.app/) grew to over 500 posts written by more than 20 authors. In 2020 we made this tool available as [a template repository on GitHub](https://github.com/x-govuk/govuk-design-history) so that other teams could quickly begin writing their own design histories. The tool has since been adopted by other teams, both inside the DfE and across government.

Reviewer: Suggest spelling out the department name in full, or defining the acronym earlier. Also I don’t think it’s common to use “the” with department acronyms?

```suggestion
With this foundation in place, [the design history](https://bat-design-history.netlify.app/) grew to over 500 posts written by more than 20 authors. In 2020 we made this tool available as [a template repository on GitHub](https://github.com/x-govuk/govuk-design-history) so that other teams could quickly begin writing their own design histories. The tool has since been adopted by other teams, both inside the Department for Education and across government.
```

Markdown written using [an extended syntax](https://x-govuk.github.io/govuk-eleventy-plugin/markdown-advanced/) is also supported, meaning authors can easily include tables, abbreviations, footnotes and code examples in their posts, too.

Knowing that teams often need to write documentation for internal users, or build websites that don’t meet [the criteria for using the GOV.UK logo and GDS Transport typeface](https://www.gov.uk/service-manual/design/making-your-service-look-like-govuk), the plugin enables the following can be customised:

* Organisation name and logo
* Font family
* Brand colour
* Licence and copyright information
* Site icons and Open Graph share images

![Examples of customised sites using the plugin](/assets/posts/govuk-eleventy-plugin/customisation-examples.png)

Sites can be published to any number of hosting platforms, including [GitHub Pages](https://pages.github.com), [Netlify](https://www.netlify.com) and [Heroku](https://www.heroku.com).

## Get started

The plugin is already being used by different teams across government who have built marketing pages, blogs, glossaries – and design histories, of course.

> Get started with the plugin by following the [instructions on the documentation site](https://x-govuk.github.io/govuk-eleventy-plugin/get-started). This project [is open source](https://github.com/x-govuk/govuk-eleventy-plugin) and contributions are welcome. If you come across an issue or have an idea for a feature, please [file an issue](https://github.com/x-govuk/govuk-eleventy-plugin/issues). We’re excited to see how you use this tool.

Reviewer: The GitHub UI seems to use “submit”. Although “post” might be a more user friendly term?

```suggestion
Get started with the plugin by following the [instructions on the documentation site](https://x-govuk.github.io/govuk-eleventy-plugin/get-started). This project [is open source](https://github.com/x-govuk/govuk-eleventy-plugin) and contributions are welcome. If you come across an issue or have an idea for a feature, please [submit an issue](https://github.com/x-govuk/govuk-eleventy-plugin/issues). We’re excited to see how you use this tool.
```
